pandas/core/generic.py

```diff
@@ -4287,6 +4287,39 @@ def _clip_with_scalar(self, lower, upper, inplace=False):
         else:
             return result
 
+    def _clip_with_one_bound(self, threshold, method, axis, inplace):
+
+        if np.any(isnull(threshold)):
+            raise ValueError("Cannot use an NA value as a clip threshold")
+
+        # method is self.le for upper bound and self.ge for lower bound
+        if is_scalar(threshold) and is_number(threshold):
```

Comment:
I fell like there are multiple lines of common code of `clip_upper` and `clip_lower` and this fix will make even more. Thus, I moved them form `clip_upper` and `clip_lower` to a new method `_clip_with_one_bound`. This is my first time to contribute code to a open-source project and I am not sure if I should do this. If I shouldn't, let me know how I should fix. Thanks.

The issue is that if you pass a list to `clip`'s lower or upper argument, an error will be raised, because it will pass lower and upper to the argument other of `where` but argument other doesn't take like or numpy array.